### PR TITLE
Fix segfault in debug mode

### PR DESCRIPTION
**Pull request by jld3103**
Uses the official engines in debug mode (like previously).

### CODE REVIEW

File: internal/enginecache/cache.go [hunk 1]
```diff
@@ -237,29 +237,109 @@ func ValidateOrUpdateEngine(targetOS, cachePath, requiredEngineVersion string, m
...
-		file += "linux"
-	case "darwin":
-		file += "macosx"
-	case "windows":
-		file += "windows"
-	}
-	file += fmt.Sprintf("_x64-host_%s.zip", mode.Name)
-	engineDownloadURL := fmt.Sprintf("https://github.com/flutter-rs/engine-builds/releases/download/f-%s/%s", requiredEngineVersion, file)
+	if mode == build.DebugMode {
+		targetedDomain := "https://storage.googleapis.com"
```
Comment (GeertJohan) [inline internal/enginecache/cache.go:241]: Didn't we have a function that gives the correct storage URI based on whether the user is behind CN firewall?

File: cmd/build.go [hunk 2]
```diff
@@ -639,7 +639,11 @@ func buildEnv(targetOS string, engineCachePath string) []string {
 	switch targetOS {
 	case "darwin":
 		cgoLdflags += fmt.Sprintf(" -L%s -L%s", engineCachePath, outputDirPath)
-		cgoLdflags += fmt.Sprintf(" -lflutter_engine -Wl,-rpath,.")
+		if buildOrRunMode == build.DebugMode {
```
Comment (GeertJohan) [inline cmd/build.go:642]: Maybe needs a comment about why this is necessary and how this works?

File: internal/enginecache/cache.go (hunk 1 above)
Comment (jld3103) [reply internal/enginecache/cache.go:241]: Only for flutter, becase they provide such service. I readded it below

**Comment by jld3103** [comment]
This should work now. Some ld flags were still wrong. I tested it on a MacOS VM in debug and release

**Comment by jld3103** [comment]
@janezstupar @jonas1-4 can you test this PR to confirm it fixes everything?

**Comment by JanezStupar** [comment]
> @JanezStupar @jonas1-4 can you test this PR to confirm it fixes everything?

It works fine on my end!

**Comment by jld3103** [comment]
Great

**Comment by jonas1-4** [comment]
Everything works fine

**Pull request merged**